model/src/main/proto/screens.proto

```diff
@@ -152,6 +152,9 @@ enum ScreenName {
 
   // Screen name value for the scenario when the policies activity is visible to the user.
   POLICIES_ACTIVITY = 47;
+
+  // A generic foreground screen value for logging periodic metrics like cpu usage.
+  FOREGROUND_SCREEN = 48;
```

Comment:
Why not just log the current screen, instead? It feels a bit odd to have a separate "foreground_screen" when technically all non-background screens are considered "in the foreground."

Reply:
The problem here is, current screen values are activity related and thus not present in the ApplicationLifecycleObserver where we keep track of background vs foreground changes. I've mentioned this as an upside to the "Possible Alternative" in the PR description above.

Reply:
Update -- we can disregard this now. I've used the activityLifecycleObserver to get currentScreen value and it works.

Reply:
Is there a way to meet in the middle by implementing a single class that's both an application listener & activity lifecycle observer?

Reply:
done. 
for now, I've not deleted the activityLifecycleObserver class -- will delete after high level approach is green flagged.

Reply:
update: this value (FOREGROUND_SCREEN) still persists following a change of approach.